xen/arch/arm/tee/ffa_partinfo.c

```diff
@@ -176,9 +193,75 @@ int32_t ffa_handle_partition_info_get(uint32_t w1, uint32_t w2, uint32_t w3,
         }
     }
 
-    *count = num_ffa_sp;
+    if ( ffa_fw_support_fid(FFA_PARTITION_INFO_GET) )
+    {
+        ffa_hyp_rx_release();
+        spin_unlock(&ffa_rx_buffer_lock);
+    }
+
+    *count = num_ffa_sp + num_ffa_vm;
     *size = dst_size;
 
+    /* add the VM informations if any */
+    if ( !num_ffa_vm )
+        return FFA_RET_OK;
+
+    rcu_read_lock(&domlist_read_lock);
+    for_each_domain( dom )
+    {
+        struct ffa_ctx *vm = dom->arch.tee;
+
+        /*
+         * we do not add the VM calling to the list and only VMs with
+         * FF-A support
+         */
+        if (dom != d && vm != NULL && vm->guest_vers != 0)
+        {
+            /*
+             * We do not have UUID info for VMs so use
+             * the 1.0 structure so that we set UUIDs to
+             * zero using memset
+             */
+            struct ffa_partition_info_1_0 srcvm;
+
+            if ( !num_ffa_vm )
+            {
+                /*
+                 * The number of domains changed since we counted them, we
+                 * can add new ones if there is enough space in the destination
+                 * buffer so check it or go out with an error.
+                 */
+                (*count)++;
+                if ( ctx->page_count * FFA_PAGE_SIZE < *count * dst_size )
+                {
+                    ret = FFA_RET_NO_MEMORY;
+                    rcu_read_unlock(&domlist_read_lock);
+                    goto out;
+                }
+                num_ffa_vm++;
+            }
+
+            srcvm.id = ffa_get_vm_id(dom);
+            srcvm.execution_context = dom->max_vcpus;
+            srcvm.partition_properties = FFA_PART_VM_PROP;
+            if ( is_64bit_domain(dom) )
+                srcvm.partition_properties |= FFA_PART_PROP_AARCH64_STATE;
+
+            memcpy(dst_buf, &srcvm, MIN(sizeof(srcvm), dst_size));
```

Comment:
`MIN(sizeof(srcvm), dst_size))` will always evaluate to `sizeof(struct ffa_partition_info_1_0)`. Do you use to `MIN()` macro to prepare for future changes?

Reply:
Yes this is making the code more error prone in the future